framework/auth/views.py

```diff
@@ -3,46 +3,49 @@
 import furl
 import httplib as http
 
-from flask import request
 import markupsafe
+from flask import request
 
 from modularodm import Q
 from modularodm.exceptions import NoResultsFound
 from modularodm.exceptions import ValidationValueError
 
-import framework.auth
-
-from framework.auth import cas, campaigns
 from framework import forms, status
-from framework.flask import redirect  # VOL-aware redirect
+from framework import auth as framework_auth
 from framework.auth import exceptions
-from framework.auth.exceptions import ExpiredTokenError, InvalidTokenError
-from framework.exceptions import HTTPError
-from framework.auth import (logout, get_user, DuplicateEmailError)
+from framework.auth import cas, campaigns
+from framework.auth import logout, get_user
+from framework.auth.exceptions import DuplicateEmailError, ExpiredTokenError, InvalidTokenError
+from framework.auth.core import generate_verification_key
 from framework.auth.decorators import collect_auth, must_be_logged_in
+from framework.auth.forms import ResendConfirmationForm, ResetPasswordForm, ForgotPasswordForm
+from framework.exceptions import HTTPError
+from framework.flask import redirect  # VOL-aware redirect
 from framework.sessions.utils import remove_sessions_for_user
-from framework.auth.forms import (
-    MergeAccountForm, RegistrationForm, ResendConfirmationForm,
-    ResetPasswordForm, ForgotPasswordForm
-)
-
-from website import settings
-from website import mails
-from website import language
-from website import security
-from website.util.time import throttle_period_expired
+from website import settings, mails, language
 from website.models import User
 from website.util import web_url_for
 from website.util.sanitize import strip_html
+from website.util.time import throttle_period_expired
 
 
 @collect_auth
-def reset_password(auth, **kwargs):
+def reset_password(auth, verification_key=None, **kwargs):
+    """
+    View for user to reset password. User submits new password and OSF attempts to reset it and automatically log the
+    user back in or return respective error.
+    HTTP Method: GET, POST
+
+    :raises: HTTPError(http.BAD_REQUEST) if verification_key is invalid
+    """
+
+    # If user is already logged in, log user out
     if auth.logged_in:
         return auth_logout(redirect_url=request.url)
-    verification_key = kwargs['verification_key']
+
     form = ResetPasswordForm(request.form)
 
+    # Check if request bears a valid verification_key
     user_obj = get_user(verification_key=verification_key)
```

Comment:
@brianjgeiger, @cslzchen - we should require `verification_key` + `username` + `expiration` for better security, also matches existing functionality of the CAS `/login` endpoint.